src/tools/info/proto_info.c

```diff
@@ -134,13 +138,33 @@ void print_ucp_info(int print_opts, ucs_config_print_flags_t print_flags,
         goto out_release_config;
     }
 
+    if (mem_size != NULL) {
+        ucs_str_to_memunits(mem_size, &mem_size_value, NULL);
+        mem_params.field_mask = UCP_MEM_MAP_PARAM_FIELD_ADDRESS |
+                                UCP_MEM_MAP_PARAM_FIELD_LENGTH  |
+                                UCP_MEM_MAP_PARAM_FIELD_FLAGS;
+        mem_params.address    = NULL;
+        mem_params.length     = mem_size_value;
+        mem_params.flags      = UCP_MEM_MAP_ALLOCATE;
+
+        status = ucp_mem_map(context, &mem_params, &memh);
+        if (status != UCS_OK) {
+            printf("<Failed to map memory of size %s>\n", mem_size);
+            goto out_cleanup_context;
+        }
+
+        if (print_opts & PRINT_MEM_MAP) {
+            ucp_mem_print_info(memh, context, stdout);
+        }
```

Comment:
we can release the memory immediately after allocating it. and put all mmap/munmap code into print_mem_info